### PR TITLE
Update survey deletion behavior.

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java [hunk 1]
```diff
@@ -532,7 +552,7 @@ public void workerCanGetSurveys() throws Exception {
         assertKeysEqual(survey1aKeys, survey1aAgain);
 
         // We only expect the most recently published versions, namely 1b and 2b.
-        SurveyList surveyResourceList = workerApi.getAllPublishedSurveys(IntegTestUtils.STUDY_ID).execute().body();
+        SurveyList surveyResourceList = workerApi.getAllPublishedSurveys(IntegTestUtils.STUDY_ID, false).execute().body();
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java]: Test for includeDeleted=true?

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java [hunk 2]
```diff
@@ -757,6 +781,57 @@ public void displayActionsInAfterRulesValidated() throws Exception {
         }
     }
     
+    @Test
+    public void canRetrieveDeletedSurveys() throws Exception {
+        SurveysApi surveysApi = developer.getClient(SurveysApi.class);
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java]: What about getPublishedSurveyVersion()? getMostRecentSurveyVersion()?

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java [hunk 3]
```diff
@@ -284,8 +284,19 @@ public void getAllVersionsOfASurvey() throws Exception {
...
+        
+        // verify includeDeleted
+        Survey oneVersion = surveyList.getItems().get(0); 
+        surveysApi.deleteSurvey(oneVersion.getGuid(), oneVersion.getCreatedOn(), false).execute();
+        
+        assertEquals(surveyList.getItems().size(),
+                surveysApi.getAllVersionsOfSurvey(key.getGuid(), true).execute().body().getItems().size());
+        
+        assertNotEquals(surveyList.getItems().size(),
+                surveysApi.getAllVersionsOfSurvey(key.getGuid(), false).execute().body().getItems().size());
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java]: We happen to know that the expected counts here are 2 and 1 respectively. Would it be better to just use thocse constants?

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java (hunk 1 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java]: Added

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java (hunk 2 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java]: These should not take an includeDeleted flag; they should resolve to the most recent version or most recently published version that is not (logically) deleted. Removed the flag from the SDK (see new PR, the first PR has been merged) and adjusted the tests here to reflect this. Note the server did not have this flag in its routes or other code for these calls.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java [hunk 4]
```diff
@@ -757,6 +786,118 @@ public void displayActionsInAfterRulesValidated() throws Exception {
...
+        surveysApi.publishSurvey(keys2.getGuid(), keys2.getCreatedOn(), false).execute();
+        surveysApi.deleteSurvey(keys2.getGuid(), keys2.getCreatedOn(), false).execute();
+        
+        Thread.sleep(1000);
+        anyDeleted(surveysApi.getPublishedSurveys(true));
+        noneDeleted(surveysApi.getPublishedSurveys(false));
+        
+        try {
+            surveysApi.getPublishedSurveys(true).execute();
+        } catch(EntityNotFoundException e) {
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java]: Need a fail() here. Otherwise, if this doesn't throw, we'll never know.

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java (hunk 4 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/SurveyTest.java]: Added here (and one other place in test).

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**